packages/squiggle-lang/src/value/index.ts

```diff
@@ -135,17 +178,46 @@ export type LabeledDistribution = {
   distribution: BaseDist;
 };
 
-export type Plot = {
-  distributions: LabeledDistribution[];
-};
-
-class VPlot {
+export type Plot =
+  | {
+      type: "distributions";
+      distributions: LabeledDistribution[];
+    }
+  | {
+      type: "fn";
+      fn: Lambda;
+      min: number;
+      max: number;
+    };
+
+class VPlot implements Indexable {
   readonly type = "Plot" as const;
   constructor(public value: Plot) {}
+
   toString() {
-    return `Plot containing ${this.value.distributions
-      .map((x) => x.name)
-      .join(", ")}`;
+    switch (this.value.type) {
+      case "distributions":
+        return `Plot containing ${this.value.distributions
+          .map((x) => x.name)
+          .join(", ")}`;
+      case "fn":
+        return `Plot for function ${this.value.fn}`;
+      default:
+        // never happens
+        return "Unknown plot";
```

Comment:
Maybe throw error

Reply:
Changed to explicit return type, typescript should handle the exhaustiveness check for us.